src/main.rs

```diff
@@ -0,0 +1,169 @@
+use typedb_client::{
+    concept::{Attribute, Concept, Value},
+    Connection, DatabaseManager, Session,
+    SessionType::{Data, Schema},
+    TransactionType::{Read, Write}
+};
+use futures::{StreamExt};
+use  std::fs;
+use text_io::read;
+
+const TEST_DATABASE: &str = "menuDB";
+ 
+fn new_core_connection() -> typedb_client::Result<Connection> {
+    Connection::new_plaintext("localhost:1729")
+}
+
+async fn load_data(connection: Connection)->std::io::Result<()>{
+    let data=fs::read_to_string("./src/data.tql")?;    
+    let databases = DatabaseManager::new(connection.clone());
+    // insert data
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Write).await.unwrap();
+    let _ = transaction.query().insert(data.as_str());
+    transaction.commit().await.unwrap();
+    println!("\nData Loaded\n");
+    Ok(())
+}
+
+async fn load_schema(connection: Connection)->std::io::Result<()>{
+    let schema = fs::read_to_string("./src/schema.tql")?;
+    let databases = DatabaseManager::new(connection.clone());
+    if databases.contains(TEST_DATABASE).await.unwrap()==false {
+        databases.create(TEST_DATABASE).await.unwrap();
+        // define schema
+        let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Schema).await.unwrap();
+        let transaction = session.transaction(Write).await.unwrap();
+        transaction.query().define(schema.as_str()).await.unwrap();
+        transaction.commit().await.unwrap();
+        session.force_close().unwrap();
+        
+        // load data for the first
+        load_data(connection.clone()).await?;
```

Comment:
Having `load_data` contained within `load_schema` might be confusing in some cases. It would be better if we called `load_data` alongside `load_schema`, and checking at the top level if the database already exists. 

I think it's OK if we use the database already existing as a heuristic to not write the schema or the data again, but let's bring that out to the top level of the logic of the program (within your `mymain`).

Reply:
I am confused here. Lets say I do following steps:
1. Use Load_schema() to create database and define schema.
2. Call load_data alongside load_schema to check if database is defined.

But If the database already exists due to step 1, then the data from data.tql will not get inserted. I was trying to do it at top level but it was not working. Please check this [commit ](https://github.com/Might1331/restaurant-menu-data-model/pull/1/commits/7efce367b5f35e58b306068d2eaf23faa303091c), here I have load_data next to load_schema and it is wrong as in the data will never get inserted.

Reply:
> I am confused here. Lets say I do following steps:
> 
>1. Use Load_schema() to create database and define schema.
> 
>2. Call load_data alongside load_schema to check if database is defined.
> 
> 
> But If the database already exists due to step 1, then the data from data.tql will not get inserted. I was trying to do it at top level but it was not working. Please check this [commit ](https://github.com/Might1331/restaurant-menu-data-model/pull/1/commits/7efce367b5f35e58b306068d2eaf23faa303091c), here I have load_data next to load_schema and it is wrong as in the data will never get inserted.

I'm thinking something like:
```rust
if !database_exists {
    create_database();
    load_schema();
    load_data();
}
```